hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanActionExecutor.java

```diff
@@ -146,10 +179,12 @@ HoodieCleanerPlan requestClean(HoodieEngineContext context) {
    */
   protected Option<HoodieCleanerPlan> requestClean(String startCleanTime) {
     final HoodieCleanerPlan cleanerPlan = requestClean(context);
-    if ((cleanerPlan.getFilePathsToBeDeletedPerPartition() != null)
+    // Create a clean request contains the cleaner plan if:
+    // - ALLOW_EMPTY_CLEAN_COMMITS is true
+    // - or the list of the file paths to be deleted is not empty
+    if (config.allowEmptyCleanCommits() || (cleanerPlan.getFilePathsToBeDeletedPerPartition() != null
```

Comment:
we should probably add another check to validate that ALLOW_EMPTY_CLEAN_COMMITS is set to true only if incremental cleaning is enabled.